icu4c/source/test/fuzzer/uprop_fuzzer.cpp

```diff
@@ -0,0 +1,44 @@
+// © 2023 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// Fuzzer for ICU Unicode Property.
+
+#include <cstring>
+
+#include "fuzzer_utils.h"
+
+#include "unicode/uchar.h"
+#include "unicode/locid.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+    UProperty prop;
+    UChar32 c32;
+
+    if (size < sizeof(prop) + sizeof(c32)) return 0;
+
+    icu::StringPiece fuzzData(reinterpret_cast<const char *>(data), size);
+
+    std::memcpy(&prop, fuzzData.data(), sizeof(prop));
```

Comment:
I'm unclear how this is supposed to work. The fuzzer API takes `data` as input as a byte array. [UProperty](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/uchar_8h.html#ae40d616419e74ecc7c80a9febab03199) is an enum.

In order to uniformly randomly testing binary properties, it would be good to:
- get a random integer from the fuzzer (not sure if the fuzz test author needs to convert fuzzer's byte array input into an integer or the fuzzer API allows random ints as input)
- do some modulo arithmetic between [UCHAR_BINARY_START](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/uchar_8h.html#ae40d616419e74ecc7c80a9febab03199aa0043677ce563a0048e0f4c3b2383869) AND UCHAR_BINARY_LIMIT
- Do the same with UCHAR_INT_START and UCHAR_INT_LIMIT
- Only call the APIs that belong to the correct property type (ex: only call `u_getIntPropertyValue` if the property is an integer/enumerated property, etc., only call APIs for binary properties when you know `prop` is a binary property)

Reply:
I don't know-- it seems like there's value in passing out-of-range (or otherwise invalid) values to the API and making sure that nothing evil happens as a result...

Reply:
> * Only call the APIs that belong to the correct property type (ex: only call `u_getIntPropertyValue` if the property is an integer/enumerated property, etc., only call APIs for binary properties when you know `prop` is a binary property)

But... what fuzzer try to test is if somehow wrong value get pass in, the API will check it is out of range and return error instead of crash. If we constraint the input to the "valid range" instead of "possible range" then it lost the meaning of using fuzzer, right? Hackers will try to attack those out of valid range value to break our code and fuzzer is trying to test against that as an evil hacker.

Reply:
Yep, you're totally right. I take back my comment. Carry on...